hack/verify-flags-underscore.py

```diff
@@ -0,0 +1,210 @@
+def is_binary(pathname):
+    """Return true if the given filename is binary.
+    @raise EnvironmentError: if the file does not exist or cannot be accessed.
+    @attention: found @ http://bytes.com/topic/python/answers/21222-determine-file-type-binary-text on 6/08/2010
+    @author: Trent Mick <TrentM@ActiveState.com>
+    @author: Jorge Orpinel <jorge@orpinel.com>"""
+    try:
+        f = open(pathname, 'r')
+        CHUNKSIZE = 1024
+        while 1:
+            chunk = f.read(CHUNKSIZE)
+            if '\0' in chunk: # found null byte
+                return True
+            if len(chunk) < CHUNKSIZE:
+                break # done
+    except:
+        return True
+    finally:
+        f.close()
+    return False
+
+def get_all_files(rootdir):
+    all_files = []
+    for root, dirs, files in os.walk(rootdir):
+        # don't visit certain dirs
+        if 'Godeps' in dirs:
+            dirs.remove('Godeps')
+        if 'third_party' in dirs:
+            dirs.remove('third_party')
+        if '.git' in dirs:
+            dirs.remove('.git')
+        if 'exceptions.txt' in files:
+            files.remove('exceptions.txt')
+        if 'known-flags.txt' in files:
+            files.remove('known-flags.txt')
+
+        for name in files:
+            if name.endswith(".svg"):
+                continue
+            if name.endswith(".gliffy"):
+                continue
+            pathname = os.path.join(root, name)
+            if is_binary(pathname):
+                continue
+            all_files.append(pathname)
+    return all_files
+
+def normalize_files(rootdir, files):
+    newfiles = []
+    a = ['Godeps', 'third_party', 'exceptions.txt', 'known-flags.txt']
+    for f in files:
+        if any(x in f for x in a):
+            continue
+        if f.endswith(".svg"):
+            continue
+        if f.endswith(".gliffy"):
+            continue
+        newfiles.append(f)
+    for i, f in enumerate(newfiles):
+        if not os.path.isabs(f):
+            newfiles[i] = os.path.join(rootdir, f)
+    return newfiles
+
+def line_has_bad_flag(line, flagre):
+    m  = flagre.search(line)
+    if not m:
+        return False
+    if "_" in m.group(0):
+        return True
+    return False
+
+# The list of files might not be the whole repo. If someone only changed a
+# couple of files we don't want to run all of the golang files looking for
+# flags. Instead load the list of flags from hack/verify-flags/known-flags.txt
+# If running the golang files finds a new flag not in that file, return an
+# error and tell the user to add the flag to the flag list.
+def get_flags(rootdir, files):
+    # use a dict for uniqueness
+    flags = {}
+
+    # preload the 'known' flags
+    pathname = os.path.join(rootdir, "hack/verify-flags/known-flags.txt")
+    f = open(pathname, 'r')
+    for line in f.read().splitlines():
+        flags[line] = True
```

Comment:
You could use a `set` instead if you need a unique container without caring about having it map to something.